dali/test/python/test_operator_jpeg_compression_distortion.py

```diff
@@ -24,47 +24,93 @@
 test_data_root = os.environ['DALI_EXTRA_PATH']
 images_dir = os.path.join(test_data_root, 'db', 'single', 'tiff')
 dump_images = False
+sequence_length = 10
 
-def _testimpl_jpeg_compression_distortion(batch_size, device, quality):
-  @pipeline_def()
+class InputImagesIter(object):
+  def __init__(self, batch_size, sequence_length):
+    self.batch_size = batch_size
+    self.sequence_length = sequence_length
+    with open(os.path.join(images_dir, 'image_list.txt')) as file:
+      self.files = [line.rstrip() for line in file if line != '']
+
+  def __iter__(self):
+    self.i = 0
+    return self
+
+  def __next__(self):
+    batch = []
+    for _ in range(self.batch_size):
+      in_img = None
+      # Skip input image if format isn't supported by OpenCV
+      while in_img is None:
+        filename, label = self.files[self.i].split(' ')
+        in_img = cv2.imread(os.path.join(images_dir, filename))
+        self.i = (self.i + 1) % len(self.files)
+        # Convert to rgb, to match dali channel order
+      rgb = cv2.cvtColor(in_img, cv2.COLOR_BGR2RGB)
+      batch.append(np.stack([rgb] * self.sequence_length))
+    return batch
+
+def _comapare_to_cv_distortion(in_img, out_img, q, no):
+  bgr = cv2.cvtColor(in_img, cv2.COLOR_RGB2BGR)
+  encode_params = [int(cv2.IMWRITE_JPEG_QUALITY), q]
+  _, encoded_img = cv2.imencode('.jpg', bgr, params=encode_params)
+
+  decoded_img_bgr = cv2.imdecode(encoded_img, cv2.IMREAD_COLOR)
+  decoded_img = cv2.cvtColor(decoded_img_bgr, cv2.COLOR_BGR2RGB)
+
+  if dump_images:
+    i, j = no
+    cv2.imwrite(f"./reference_q{q}_sample{i}_{j}.bmp", cv2.cvtColor(decoded_img, cv2.COLOR_BGR2RGB))
+    cv2.imwrite(f"./output_q{q}_sample{i}_{j}.bmp", cv2.cvtColor(out_img, cv2.COLOR_BGR2RGB))
+
+    diff = cv2.absdiff(out_img, decoded_img)
+    assert np.average(diff) < 5, f"Absolute difference with the reference is too big: {np.average(diff)}"
```

Comment:
I'm afraid that now you assert only if `dump_images==True`.
```suggestion
  diff = cv2.absdiff(out_img, decoded_img)
  assert np.average(diff) < 5, f"Absolute difference with the reference is too big: {np.average(diff)}"
```

Reply:
Yes I do.

Reply:
Fixed

Reply:
I'm afraid you have missed that.